docs/interfaces/datastore-service.txt

```diff
@@ -119,70 +120,69 @@ Event ModifiedFieldsEvent on topic ModifiedFields {
  * - parameter `position`: Optional, if given reads the data to this position.
- * - parameter `mapped_fields`: List of fields, that should onl be present in the response.
+ * - parameter `mapped_fields`: List of fields that should only be present in the response.
+ * - parameter `get_deleted_models`: Optional, defines which models to return
+ *    - DeletedModelsBehaviour.NO_DELETED:   (Default) only non-deleted models are returned.
+ *                                           get throws a ModelDoesNotExist error if the given
+ *                                           model is deleted.
+ *    - DeletedModelsBehaviour.ONLY_DELETED: only deleted models are returned. get throws
+ *                                           a ModelNotDeleted if the given model is not deleted.
+ *    - DeletedModelsBehaviour.ALL_MODELS:   all models are returned
  * - All operations adds the fields `meta_position` and `meta_deleted` to the models.
  * - The InvalidFormat exception can always be thrown, if the requested formats are
  *   wrong, including something like empty collections, ...
  */
 
-/**
- * Returns a model. Deleted models are not returned (and handled as a ModelDoesNotExit)
- *
- * @throws ModelDoesNotExist
- * @throws InvalidFormat
- */
-get(model: Fqid, position?: Position, mapped_fields?: fields[]): Partial<Model>;
+Enum DeletedModelsBehaviour {
+    NO_DELETED,
+    ONLY_DELETED,
+    ALL_MODELS
+}
 
 /**
- * Analogous to `get`, but also finds deleted models (see `meta_deleted` in the model)
+ * Returns a model by fqid.
  *
  * @throws ModelDoesNotExist
  * @throws InvalidFormat
  */
-getWithDeleted(model: Fqid, position?: Position, mapped_fields?: Field[]): Partial<Model>;
-
-/**
- * Returns multiple (non-deleted) models of a collection. If one id is not found, it is
- * not included in the response instead of throwing a ModelDoesNotExist.
- */
-getMany(collection: Collection, ids: Id[], position?: Position, mapped_fields?: Field[]): Partial<Model>[];
+get(fqid: Fqid, mapped_fields?: Field[], position?: Position, get_deleted_models?: DeletedModelsBehaviour): Partial<Model>;
 
 /**
- * Analogous to `getMany`, but includes deleted instances.
+ * Returns multiple models.
+ * Can either be called with a list of fqids (if all fields are needed/wanted or if the
+ * same fields of all objects are needed) or with a list of specific request objects
+ * that map a collection to the needed ids and fields. If both the lower and the higher
+ * level mapped_fields are given, the higher level one is merged into all lower level
+ * ones.
+ * If an id is not found, it is not included in the response instead of throwing a
+ * ModelDoesNotExist.
  *
  * @throws InvalidFormat
  */
-getManyWithDeleted(collection: Collection, ids: Id[], position?: Position, mapped_fields?: Field[]): Partial<Model>[];
+getMany(requests: Fqid[] | GetManyRequest[], mapped_fields?: Field[], position?: Position, get_deleted_models?: DeletedModelsBehaviour): Partial<Model>[];
```

Comment:
I do not like the extra possibility to give fqids. Currently this request is possible:
```
getMany([{collection:"a", ids: [1, 2], mapped_fields: ["f"]}], mapped_fields: ["g"]);
```
I would prefet the following suggestion:
```suggestion
getMany(requests: GetManyRequest[], position?: Position, get_deleted_models?: DeletedModelsBehaviour): Partial<Model>[];
```

Reply:
You are mixing 2 things here: the possibility to give fqids instead of request objects and the possible two mapped_fields.
1) The idea was that if there was a situation where you want to get models of different collections, but a common field of all of them, you could just pass in the fqids and the mapped fields. Of course that's not the typical use case. If you want we can remove the possibility of fqids.
2) The 2 mapped_fields are more useful in my opinion and can exist without the possibility to pass in fqids, if one wants to get a common field of multiple models.

Reply:
OK